### PR TITLE
Add examples to documentation, small compiler change

### PR DESCRIPTION
**Pull request by fluffysquirrels**
I'm happy to split the commits into different pull requests if it would help.

Tested with `make check` and buliding my own project.

As discussed in [issue 13](https://github.com/opensourcerouting/c-capnproto/issues/13).

### CODE REVIEW

**Comment by fluffysquirrels** [comment]
Whoops, looks like the clang build failed. I need to go out, but I'll fix this later.

File: Makefile.am [hunk 1]
```diff
@@ -56,6 +56,8 @@ check_PROGRAMS += \
 capn_test_SOURCES = \
 	tests/capn-test.cpp \
 	tests/capn-stream-test.cpp \
+	tests/example-test.cpp \
+	tests/example.capnp.c \
 	compiler/test.capnp.c \
 	compiler/schema-test.cpp \
 	compiler/schema.capnp.c
```
Comment (eqvinox) [inline Makefile.am]: need to add tests/example.capnp and tests/example.capnp.h in Makefile.am so they will be included in tarballs (make dist) - example.capnp needs to go to `EXTRA_DIST` and example.capnp.h into `noinst_HEADERS`

Other than that, looks good 👍

Comment (fluffysquirrels) [inline Makefile.am]: Fixed. Sorry, first time with automake.
Output of `make dist; tar -tvf c-capnproto-0.2.tar.gz | sort -f6`:
```
drwxrwxr-x 1000/1000         0 2017-03-24 17:58 c-capnproto-0.2/tests/
-rw-rw-r-- 1000/1000      7752 2017-03-23 00:36 c-capnproto-0.2/tests/capn-stream-test.cpp
-rw-rw-r-- 1000/1000     14902 2017-03-23 00:36 c-capnproto-0.2/tests/capn-test.cpp
-rw-rw-r-- 1000/1000       178 2017-03-23 20:18 c-capnproto-0.2/tests/example.capnp
-rw-rw-r-- 1000/1000      1826 2017-03-23 20:18 c-capnproto-0.2/tests/example.capnp.c
-rw-rw-r-- 1000/1000      1179 2017-03-23 20:18 c-capnproto-0.2/tests/example.capnp.h
-rw-rw-r-- 1000/1000      1818 2017-03-24 02:31 c-capnproto-0.2/tests/example-test.cpp
```
I noticed the other pre-existing `.capnp` files weren't in `EXTRA_DIST` but when I tried adding them they broke `make dist` with this output:
```
 % make dist
 cd . && /bin/bash /home/alex/Code/stm32/modules/arm-gcc-toolchain/linux/c-capnproto/source/ac/missing automake-1.15 --foreign Makefile
 cd . && /bin/bash ./config.status Makefile depfiles
config.status: creating Makefile
config.status: executing depfiles commands
make  dist-gzip am__post_remove_distdir='@:'
make[1]: Entering directory '/home/alex/Code/stm32/modules/arm-gcc-toolchain/linux/c-capnproto/source'
gcc -g -O2    compiler/test.capnp.c   -o compiler/test.capnp
In file included from compiler/test.capnp.c:1:0:
compiler/test.capnp.h:4:21: fatal error: capnp_c.h: No such file or directory
compilation terminated.
<builtin>: recipe for target 'compiler/test.capnp' failed
make[1]: *** [compiler/test.capnp] Error 1
make[1]: Leaving directory '/home/alex/Code/stm32/modules/arm-gcc-toolchain/linux/c-capnproto/source'
Makefile:1485: recipe for target 'dist' failed
make: *** [dist] Error 2
```
Any ideas how to fix this?

Comment (eqvinox) [inline Makefile.am]: hm, gimme a day or two to look at this, got some bits ahead of it in the priority queue :/

Comment (fluffysquirrels) [inline Makefile.am]: Of course, no problem :)

It looks like for some `.capnp` files Make is trying to build `${x}.capnp` from `${x}.capnp.c`, but I haven't dug in further.

Thanks for the repo access.

Comment (fluffysquirrels) [inline Makefile.am]: Doing something unrelated I just came across [this fix](https://github.com/opensourcerouting/c-capnproto/blob/master/compiler/c.capnp#L24-L32) in the source. I can try and use it.

Comment (fluffysquirrels) [inline Makefile.am]: I fixed it with a simple Makefile rule: `%.capnp: ;`. Documented this in README.md and where the previous workaround was documented.

**Comment by eqvinox** [comment]
Sorry, this took a while (we've just had a major release) - merged & thanks a ton!

**Comment by fluffysquirrels** [comment]
@eqvinox No problem at all about the delay. I hope your release went well :)

Happy to help with the project where I can.

**Pull request merged**